product_taxes_group/model/product_template.py

```diff
@@ -71,49 +72,28 @@ def fields_view_get(
         return res
 
     # Custom Section
-    def check_coherent_vals(self, ids, vals):
+    def check_coherent_vals(self, vals):
         """If tax group is defined, set the according taxes to the product(s);
         Otherwise, find the correct tax group, depending of the taxes, or
         create a new one, if no one are found.
         """
-        tg_obj = self.env['tax.group']
         if vals.get('tax_group_id', False):
             # update or replace 'taxes_id' and 'supplier_taxes_id'
-            tg = tg_obj.browse(vals['tax_group_id'])
-            vals['supplier_taxes_id'] = [[6, 0, [
-                x.id for x in tg.supplier_tax_ids]]]
-            vals['taxes_id'] = [[6, 0, [
-                x.id for x in tg.customer_tax_ids]]]
+            tax_vals = {
+                'supplier_taxes_id': [[6, 0, [
+                    x.id for x in self.tax_group_id.supplier_tax_ids]]],
```

Comment:
Don't you have to sudo to access to all the supplier taxes of the current tax_group_id ? (that is your objective it isn't ?)